### ISSUE TITLE
dotnet: detect large switch statements towards identifying flattened .NET code

### ISSUE THREAD

**Issue by mike-hunhoff** [mandiant/capa#1455]
The CIL [`Switch` instruction](https://learn.microsoft.com/en-us/dotnet/api/system.reflection.emit.opcodes.switch?view=net-8.0) encodes the number of cases directly in the instruction so we can detect large switch statements. We often see large switch statements used for control flow flattening (CFF) so this could be a way for capa to detect obfuscated .NET code.

Tasks:
- [ ] research common .NET CFF techniques to determine prevalence of large switch statements (vs. large `if-else` statements, etc.) and to determine a reliable switch case count threshold that indicates flattened .NET code
- [ ] add a new `characteristic` feature e.g. `characteristic('contains flattened code')`

**Comment by mr-tz** [comment]
There's probably other heuristics we can take into account, like odd numbers and operations being used.

I wonder though if this would be one of these characteristics that only map to one rule. I.e. would this be useful for multiple rules?

**Comment by mike-hunhoff** [comment]
> There's probably other heuristics we can take into account, like odd numbers and operations being used.
> 
> I wonder though if this would be one of these characteristics that only map to one rule. I.e. would this be useful for multiple rules?

@mr-tz agreed - before implementing we need additional brainstorming to determine if such a characteristic has applications outside of a single rule.

**Comment by gmr-tz** [comment]
One idea could be to warn users about obfuscations and other things that mess with capa's results. Other examples:
- obfuscated strings
- run-time linked API calls
- only few or no library functions identified (could indicate more FPs)
- packed

We have something already with the limitations, so could extend or change that.

The output could be something like:
```
$ capa sample.exe
Warnings:
- sample's API calls may be obfuscated
- strings may be obfuscated
- only few library functions detected

<capa result tables>
```
